### PR TITLE
Generate documentation file for config group

### LINKED ISSUE #5097: generated docs are sometimes overridden when running full build
**Describe the bug**
When running a full build, we sometimes override a previous part of a generated doc

**Expected behavior**
We should merge extensions doc and not overriding a partial or fully generated one

**Actual behavior**
Docs are overridden - e.g when running a full build, the `quarkus-vertx-http.adoc` generated doc does not contain all the information about http configuration. 

But when build the `vertx-http` extension alone, the file in question contains all the information.

**Environment (please complete the following information):**
 - Quarkus version or git rev: problem observed in latest master 9ed64388726143c99e61db3f966373ae6dcfab88

**Additional context**
I'll come up with a fix, still investigating on ways to work around this.

### PR DESCRIPTION
**Pull request by machi1990**
**1st commit:**
Make it possible to generate a file par config group. 

**2nd commit:** 
use the generated file for `FormAuthConfig` to avoid the usage of an inexistant file which  was generating the following warning
```
asciidoctor: INFO: security-guide.adoc: line 69: optional include dropped because include file not found: target/asciidoc/generated/config/quarkus-http-auth-form.adoc 
```

**3rd commit:**
Resolves https://github.com/quarkusio/quarkus/issues/5097

### CODE REVIEW

File: docs/src/main/asciidoc/security-guide.adoc [hunk 1]
```diff
@@ -64,10 +64,6 @@ value. This cookie contains a expiry time as part of the encrypted value, so all
 clocks synchronised. At one minute intervals a new cookie will be generated with an updated expiry time if the session
 is in use.
 
-The following properties can be used to configure form based auth:
```
Comment (machi1990) [inline docs/src/main/asciidoc/security-guide.adoc]: Should we add a link to the form-auth section of all-doc-page?

Comment (gsmet) [reply docs/src/main/asciidoc/security-guide.adoc]: These properties are not generated in a smaller file?

Comment (machi1990) [reply docs/src/main/asciidoc/security-guide.adoc]: IIRC, they are not generated in a smaller file.

Comment (gsmet) [reply docs/src/main/asciidoc/security-guide.adoc]: Hmm, I don't understand. I remember having all of them in small files. Is it gone?

Comment (gsmet) [reply docs/src/main/asciidoc/security-guide.adoc]: Would it be possible to generate a file per config group too? Looks like we have quite a lot of docs where it makes sense.

Comment (machi1990) [reply docs/src/main/asciidoc/security-guide.adoc]: +1, I was leaning toward that path. I'll look at it.

Comment (FroMage) [reply docs/src/main/asciidoc/security-guide.adoc]: Why is it inexistant? I reviewed that guide last week and didn't see any inexistant file?

Comment (machi1990) [reply docs/src/main/asciidoc/security-guide.adoc]: The `FormAuthConfig` configuration is a config group, and we do not generate specific doc files for config groups. If it was previously there, maybe the file being referenced here was for a config root that is no longer there.

Comment (FroMage) [reply docs/src/main/asciidoc/security-guide.adoc]: Ah, that's possible.

**Comment by machi1990** [comment]
@gsmet I have added a commit which enable the generation of config group as discussed above.

**Review by gsmet — APPROVED** [review]
I checked the output and everything looks in order. We can tune it later if needed.

**Pull request merged**